mobile/test/common/integration/base_client_integration_test.cc

```diff
@@ -158,41 +163,53 @@ std::shared_ptr<Platform::RequestHeaders> BaseClientIntegrationTest::envoyToMobi
   return std::make_shared<Platform::RequestHeaders>(builder.build());
 }
 
-void BaseClientIntegrationTest::threadRoutine(absl::Notification& engine_running) {
-  setOnEngineRunning([&]() { engine_running.Notify(); });
-  engine_ = build();
+void BaseClientIntegrationTest::threadRoutine(absl::BlockingCounter& engine_running) {
+  builder_.setOnEngineRunning([&]() { engine_running.DecrementCount(); });
+  for (auto i = num_engines_for_test_; i--;) {
+    multi_engines_.push_back(builder_.build());
```

Comment:
Note that this branch doesn't currently separate out the builder() but I've tried that in a separate branch and it didn't change the error produced.